editor/inserter/index.js

```diff
@@ -33,6 +39,27 @@ class Inserter extends Component {
 		};
 	}
 
+	componentWillUpdate( nextProps, nextState ) {
+		const { insertIndex, setInsertionPoint, clearInsertionPoint } = nextProps;
+		const { opened } = nextState;
+		if (
+			( this.state.opened !== opened || this.props.insertIndex !== insertIndex ) &&
+			Number.isInteger( insertIndex )
```

Comment:
When is this second check necessary?

Reply:
The global and end-of-content inserters don't specify the `insertIndex` prop, so this ensures the sibling insertion point is only set for the between-inserters. It _might_ be the case that `insertIndex !== undefined` could work well here. I used `Number.isInteger` elsewhere to capture `0` but not `null` (since JavaScript is a very sensible language where `null >= 0` evaluates to true 😆 )